### PR TITLE
Fix 2k3 class change + Face graphic fixes

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #759 
Fix: #2039 ???

### CODE REVIEW

File: src/window_message.cpp [hunk 1]
```diff
@@ -246,7 +246,7 @@ void Window_Message::InsertNewPage() {
 		gold_window->SetBackOpacity(GetBackOpacity());
 	}
 
-	if (!Game_Message::GetFaceName().empty()) {
+	if (IsFaceEnabled() && !Game_Message::GetFaceName().empty()) {
```
Comment (carstene1ns) [inline src/window_message.cpp]: double check.

Comment (fmatthew5876) [reply src/window_message.cpp]: Fixed, thanks!

File: src/game_actor.cpp [hunk 2]
```diff
@@ -1118,31 +1119,58 @@ const RPG::Class* Game_Actor::GetClass() const {
...
-
-	GetData().class_id = _class_id;
+void Game_Actor::ChangeClass(int new_class_id,
+		int new_level,
+		ClassChangeSkillMode new_skill,
+		ClassChangeParamMode new_param,
+		PendingMessage* pm
+		)
+{
+	Output::Debug("CLASS CHANGE %d %d %d %d", new_class_id, new_level, new_skill, new_param);
```
Comment (carstene1ns) [inline src/game_actor.cpp]: debug leftover.

Comment (fmatthew5876) [reply src/game_actor.cpp]: Thanks for catching that, fixed.

**Comment by fmatthew5876** [comment]
I added back the custom feature to always display EXP amount even at maximum level. Left a comment including correct commented out code. I personally like it better this way.

If you guys disagree we can remove it and be strictly compatible with RPG_RT.

**Comment by Ghabry** [comment]
This is hard to test because RPG\_RT behaviour makes no sense :roll_eyes:

**Comment by fmatthew5876** [comment]
https://github.com/fmatthew5876/EasyRPG-InterpreterTestGame

I have a Class test map which I used to test all the edge cases

**Comment by Ghabry** [comment]
From a code review PoV I accept this.

---------
Bug 1:
@> Change Level: Entire Party, + 30
@> Change Actor Class: [Zack], [Soldier A] with Lv: Change to 1 Skills: Learn additional, Stats: From Level 1

Menu shows EXP: 316035 / 768

Skill menu shows MP: 0 / [NOTHING] <- rendered out of bounds
-----------------
Bug 2:

@> Change Level: Entire Party, + 30
@> Change Actor Class: [Zack], to classlessLv: No change Skills: Based on Level Stats: By level
@> Change Level: Entire Party, + 30
@> Change Actor Class: [Zack], to classlessLv: No change Skills: Learn additional Stats: Halved

This shows only "Reached LV30" in RPG_RT and Reached 30, Reached 60 in Player. But our behaviour looks more sane.

**Comment by fmatthew5876** [comment]
For bug 2, we aren't strictly compatible with what level up messages we show. RPG_RT is very strange and buggy here. 

I tried to make the way we do it more sensible

**Comment by Ghabry** [comment]
I just thought about BUG 2 again because I forgot some detail:
```
@> Change Level: Entire Party, + 30 [ ] Show Message
@> Change Actor Class: [Zack], to classlessLv: No change Skills: Based on Level Stats: By level [X] Show Message
@> Change Level: Entire Party, + 30 [ ] Show Message
@> Change Actor Class: [Zack], to classlessLv: No change Skills: Learn additional Stats: Halved [X] Show Message
```
So the messages for "Change Level" were disabled why does "Change class" show any level up message then?
Question is just about whether showing nothing or both makes more sense.
At least it doesn't affect the logic, so users probably won't care except for timing edge cases.

**Comment by fmatthew5876** [comment]
Change class shows the level message when skills are changed. I think the reason rpgrt does this is because the level up message contains the actor name but the learning messages do not.

We mostly follow rpgrt except stupid edge cases where you change to the same class where rpgrt has a lot of bugs.

The behaviors are detailed in the test cases.

I will recheck this stuff later this evening to be sure

**Comment by fmatthew5876** [comment]
Fixed the EXP issue and the menu SP text.
For menus I rechecked:
- [x] Main Menu
- [x] Skills
- [x] 2k3 battle
- [x] 2k battle
- [x] ActorTarget Window from using a healing skill / item

Also rechecked your level 30/60 example. RPG_RT is buggy here and ours is more correct so left it as is.
